### PR TITLE
Prevent conflicts between service account and jwt issuers

### PR DESCRIPTION
**Pull request by enj**
/kind feature
/kind api-change
/assign liggitt aramase 
/sig auth
/triage accepted
/priority important-soon

```release-note
Conflicting issuers between JWT authenticators and service account config are now detected and fail on API server startup.  Previously such a config would run but would be inconsistently effective depending on the credential.
```

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
/lgtm

File: staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go [hunk 1]
```diff
@@ -69,52 +69,64 @@ func ValidateAuthenticationConfiguration(c *api.AuthenticationConfiguration) fie
...
 	}
 
 	return allErrs
 }
 
+func buildDisallowedIssuers(disallowedIssuers []string) sets.Set[string] {
+	return sets.New(disallowedIssuers...).Insert(
+		// always include the commonly used values that could accidentally resolve in-cluster
+		"https://kubernetes.default.svc",
+		"https://kubernetes.default.svc.cluster.local",
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go:83]: why hard-code these if they weren't used as the service account issuers?

**Comment by liggitt** [comment]
one question on the hard-coded entries, lgtm otherwise

**Pull request merged**